### PR TITLE
Fix ref by value of parameters with plus/minus

### LINKED ISSUE #2595: parameters and hypens
At present the following:

```ini
[cylc]
    [[parameters]]
        task = a-b, b-c
[scheduling]
    [[dependencies]]
        graph = """
            foo => <task=a-b>
        """
```

Will result in:

```
ERROR, parameter task out of range: task=a-b
```

### PR DESCRIPTION
**Pull request by matthewrmshin**
Syntax such as `<world=earth-moon>` can now validate.

Fix #2595.

### CODE REVIEW

File: lib/cylc/param_expand.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def expand(template, params, results, values=None):
 # To extract all parameter lists e.g. 'm,n,o' (from '<m,n,o>').
 REC_P_GROUP = re.compile(r"<(.*?)>")
 # To extract parameter name and optional offset or value e.g. 'm-1'.
-REC_P_OFFS = re.compile(r'(\w+)(?:\s*([-+=]\s*[\w]+))?')
+REC_P_OFFS = re.compile(r'(\w+)([\-\+]\d+|=%s)?' % TaskID.NAME_SUFFIX_RE)
```
Comment (oliver-sanders) [inline lib/cylc/param_expand.py]: We may need to permit arbitrary white-space between operators.

Comment (matthewrmshin) [reply lib/cylc/param_expand.py]: White space should all be removed by now, see also #2534.

Comment (oliver-sanders) [reply lib/cylc/param_expand.py]: Good. Could we sneak a little whitespace into one of the tasks in the test to make sure this doesn't get broken with #2534?

Comment (matthewrmshin) [reply lib/cylc/param_expand.py]: OK. It is unfortunate that usage of this regular expression for parsing subsections in `[runtime]` does require white-space matching.

File: tests/param_expand/01-basic.t [hunk 2]
```diff
@@ -366,4 +366,57 @@ run_ok "${TEST_NAME_BASE}-18" cylc validate "suite.rc"
...
+        [[[environment]]]
+            CC = gcc
+    [[<lang = fortran-2008>]]
+        [[[environment]]]
+            FC = gfortran
+__SUITE__
+run_ok "${TEST_NAME_BASE}-19" cylc validate --debug "suite.rc"
+cylc graph --reference 'suite.rc' >'19.graph'
+cmp_ok "${TEST_SOURCE_DIR}/${TEST_NAME_BASE}/19.graph.ref" '19.graph'
+# Note: This also demonstrates current badness of "cylc get-config"...
```
Comment (oliver-sanders) [inline tests/param_expand/01-basic.t]: Yep, see also #2392.

File: tests/param_expand/01-basic.t [hunk 3]
```diff
@@ -366,4 +366,57 @@ run_ok "${TEST_NAME_BASE}-18" cylc validate "suite.rc"
...
+# Reference by value, with -+ meta characters
+cat >'suite.rc' <<'__SUITE__'
+[cylc]
+    [[parameters]]
+        lang = c++, fortran-2008
+    [[parameter templates]]
+        lang = %(lang)s
+[scheduling]
+    [[dependencies]]
+        graph = "<lang=c++> => <lang = fortran-2008>"
```
Comment (hjoliver) [inline tests/param_expand/01-basic.t]: Many would argue against porting in this direction :grin:

Comment (matthewrmshin) [reply tests/param_expand/01-basic.t]: I didn't think about this one! Honest.

**Pull request merged**